### PR TITLE
Cleancode rules

### PR DESCRIPTION
**Pull request by tvbeek**
Type: Refactoring , follow our own cleancode rules
Issue: Part of #812
Breaking change: no

The issues that are left are:

1. src/main/php/PHPMD/Renderer/HTMLRenderer.php:586 The method reduceWhitespace has a boolean flag argument $eol, which is a certain sign of a Single Responsibility Principle violation.
2. src/main/php/PHPMD/Rule/Naming/LongClassName.php:48      Avoid using static access to class '\PHPMD\Utility\Strings' in method 'apply'.
3. src/main/php/PHPMD/Rule/Naming/LongClassName.php:62      Avoid using static access to class '\PHPMD\Utility\Strings' in method 'getSubtractSuffixList'.
4. src/main/php/PHPMD/Rule/Naming/LongVariable.php:111      Avoid using static access to class '\PHPMD\Utility\Strings' in method 'checkMaximumLength'.
5. src/main/php/PHPMD/Rule/Naming/LongVariable.php:194      Avoid using static access to class '\PHPMD\Utility\Strings' in method 'getSubtractSuffixList'.
6. src/main/php/PHPMD/Rule/Naming/ShortClassName.php:69     Avoid using static access to class '\PHPMD\Utility\Strings' in method 'getExceptionsList'.
7. src/main/php/PHPMD/RuleSetFactory.php:325        Remove error control operator '@' on line 350.
8. src/main/php/PHPMD/TextUI/Command.php:121        Remove error control operator '@' on line 127.
9. src/main/php/PHPMD/TextUI/CommandLineOptions.php:448     Remove error control operator '@' on line 464.
10. src/main/php/PHPMD/TextUI/CommandLineOptions.php:448     Remove error control operator '@' on line 474.
11. src/main/php/PHPMD/Writer/StreamWriter.php:63    Remove error control operator '@' on line 66.

point one could be split in to functions and deprecate this function. If wanted I can create another PR for that.

### CODE REVIEW

File: src/main/php/PHPMD/TextUI/CommandLineOptions.php [hunk 1]
```diff
@@ -209,6 +210,7 @@ public function __construct(array $args, array $availableRuleSets = array())
                 case '--reportfile-text':
                 case '--reportfile-xml':
                 case '--reportfile-json':
+                    $match = array();
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/TextUI/CommandLineOptions.php:213]: Dead code as preg_match fill it by reference.
```suggestion
```

File: src/main/php/PHPMD/Node/Annotations.php [hunk 2]
```diff
@@ -45,6 +45,7 @@ class Annotations
      */
     public function __construct(\PHPMD\AbstractNode $node)
     {
+        $matches = array();
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Node/Annotations.php:48]: Dead code as preg_match_all fill it by reference.
```suggestion
```

**Review by kylekatarnls — CHANGES_REQUESTED** [review]
I can't see fair reason for initializing `$match` before `preg_match*` functions.

**Comment by tvbeek** [comment]
> 
> 
> I can't see fair reason for initializing `$match` before `preg_match*` functions. PHPMD 2.9 is not supposed to raise an error for them.

I did check it with the own source code (so in fact with master)

But it is true it shouldn't create a problem, I'm thinking what the best way is to handle this.

**Comment by kylekatarnls** [comment]
This is supposed to not been seen as undefined if passed by reference:
https://github.com/phpmd/phpmd/blob/ce10831d4ddc2686c1348a98069771dd314534a8/src/main/php/PHPMD/Rule/AbstractLocalVariable.php#L287

And this actually works in unit tests.

**Comment by tvbeek** [comment]
Can you verify that you don't get an error about it? (Then we can possible close #840 )

**Comment by kylekatarnls** [comment]
@tvbeek I fixed this conflict locally, but I need you to allow "edit for maintainers" I guess I got a permission error on push.

**Comment by tvbeek** [comment]
@kylekatarnls I miss the option to verify or enable it for this PR so give you permission to my fork.

**Review by kylekatarnls — APPROVED** [review]

**Pull request merged**